src/components/embedWrapper.tsx

```diff
@@ -0,0 +1,243 @@
+			};
+		case EmbedKind.YouTube:
+			return {
+				kind: EmbedKind.YouTube,
+				id: embed.id,
+				width: embed.width.toString(),
+				height: embed.height.toString(),
+				...withDefault<Record<string, string>>({})(
+					map<string, Record<string, string>>((source) => {
+						return { source: source };
+					})(embed.source),
+				),
+				...withDefault<Record<string, string>>({})(
+					map<string, Record<string, string>>((sourceDomain) => {
+						return { sourceDomain };
+					})(embed.sourceDomain),
+				),
+				...(embed.tracking && { tracking: embed.tracking.toString() }),
+			};
+		case EmbedKind.Generic: {
+			return {
+				kind: EmbedKind.Generic,
+				...withDefault({})(
+					map<string, Record<string, string>>((alt) => {
+						return { alt };
+					})(embed.alt),
+				),
+				html: embed.html,
+				height: embed.height.toString(),
+				...(embed.mandatory && { mandatory: 'true' }),
+				...withDefault<Record<string, string>>({})(
+					map<string, Record<string, string>>((source) => {
+						return { source };
+					})(embed.source),
+				),
+				...withDefault<Record<string, string>>({})(
+					map<string, Record<string, string>>((sourceDomain) => {
+						return { sourceDomain };
+					})(embed.sourceDomain),
+				),
+				...(embed.tracking && { tracking: embed.tracking.toString() }),
+			};
+		}
+	}
+};
+
+const divElementPropsToEmbed = (container: Element): Result<string, Embed> => {
+	const parseTrackingParam = (param?: string): EmbedTracksType => {
+		switch (param) {
+			case '0':
+				return EmbedTracksType.UNKNOWN;
+			case '1':
+				return EmbedTracksType.TRACKS;
+			case '2':
+				return EmbedTracksType.DOES_NOT_TRACK;
+			default:
+				return EmbedTracksType.DOES_NOT_TRACK;
+		}
+	};
+
+	const requiredStringParam = (
+		container: Record<string, string>,
+		parameterName: string,
+	): Result<string, string> => {
+		return resultFromNullable(
+			`I can't find a '${parameterName}' field for this embed`,
+		)(container[parameterName]);
+	};
+
+	const requiredNumberParam = (
+		container: Record<string, string>,
+		parameterName: string,
+	): Result<string, number> => {
+		return andThen((value: string) => {
+			const parsedValue = Number.parseInt(value);
+
+			if (Number.isNaN(parsedValue)) {
+				return err(`${value} is not a integer`);
+			}
+
+			return ok(parsedValue);
+		})(requiredStringParam(container, parameterName));
+	};
+
+	const parseDataAttributesFromElement = (
+		container: Element,
+	): Record<string, string> => {
+		return container
+			.getAttributeNames()
+			.filter((currentKey) => currentKey.startsWith('data-'))
+			.reduce((acc, currentKey) => {
+				const newKey = currentKey
+					.replace(/^data-/g, '')
+					.replace(/-./g, (x) => x.toUpperCase()[1]);
+				const currentValue = container.getAttribute(currentKey);
+				return { ...acc, [newKey]: currentValue };
+			}, {});
+	};
+
+	const elementProps = parseDataAttributesFromElement(container);
+
+	switch (elementProps['kind'] as EmbedKind) {
```

Comment:
Can we parse the `kind` field so we can guarantee its type and/or add a `default` to this `switch`? Just want to make sure we avoid an invalid value sneaking through and accidentally returning `undefined`.